assets/src/edit-story/components/header/buttons.js

```diff
@@ -104,27 +109,21 @@ function PreviewButton() {
       // will be resolved after the story is saved.
     }
 
-    // @todo: See https://github.com/google/web-stories-wp/issues/1149. This
-    // has the effect of pushing the changes to a published story.
-    saveStory().then(() => {
-      let previewOpened = false;
-      if (popup && !popup.closed) {
-        try {
-          // The `popup.location.href` will fail if the expected window has
-          // been naviagted to a different origin.
+    // Save story directly if draft, otherwise, use auto-save.
+    const updateFunc = isDraft ? saveStory : autoSave;
+    updateFunc()
+      .then((update) => {
+        if (popup && !popup.closed) {
           if (popup.location.href) {
-            popup.location.replace(previewLink);
-            previewOpened = true;
+            // Auto-save sends an updated preview link, use that instead if available.
+            const updatedPreviewLink = update?.preview_link ?? previewLink;
+            popup.location.replace(updatedPreviewLink);
           }
-        } catch (e) {
-          // Ignore the errors. They will simply trigger the "try again"
-          // dialog.
         }
-      }
-      if (!previewOpened) {
+      })
+      .catch(() => {
```

Comment:
`setPreviewLinkToOpenViaDialog` should be called in both cases: when popup navigation fails, or when popup is not available. I.e. part of the old `previewOpened` is still needed, I think.

Reply:
If the opened preview tab would get navigated to a different URL, it would still trigger an error and the dialog open in `catch` (which was the main reason to use the `previewOpened` if I understood correctly, might be missing something though).

One difference: when the user closes the Preview tab before the request has finished, then with `previevOpened`, it displays the dialog of preview failing, with the current logic it would not.

It seemed unexpected that closing the tab purposefully meanwhile would trigger the failure dialog, however, will add it back if that's actually expected.

Reply:
Yeah. I see your point. Ok, let's run with it then.